src/Runtime/ExecutionSession.cpp

```diff
@@ -96,10 +97,27 @@ OMTensorList *ExecutionSession::run(OMTensorList *input) {
   return _entryPointFunc(input);
 }
 
-std::string ExecutionSession::inputSignature() { return _inputSignatureFunc(); }
+void ExecutionSession::setEntryPoint(const std::string &entryPointName) {
```

Comment:
If a user does not want the default entry point, and does not call `setEntryPoint`, then the `_entryPointFunc` remains null, and is unconditionally called in the `run` function. I suggest to add something similar to the code below inside the `run` function.
```C
if (!_entryPointFunc) {
     std::stringstream errStr;
     errStr << "Must set the entry point before calling run function"<< std::endl;
     throw std::runtime_error(errStr.str());
   }
```
Would do the same for the other signature functions too.

Reply:
Done for this and other functions too. Thanks!